keras_cv/training/simclr/simclr.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+
+from keras_cv.layers import preprocessing
+from keras_cv.losses import SimCLRLoss
+
+
+class SimCLR(keras.Model):
+    def __init__(
+        self,
+        encoder,
+        include_probing,
+        classes=None,
+        projection_width=128,
+        augmenter=None,
+        include_rescaling=None,
+    ):
+        super().__init__()
+
+        self.include_probing = include_probing
+        self.projection_width = projection_width
+
+        if not augmenter and not include_rescaling:
+            raise ValueError(
+                "`include_rescaling` is required when using the default augmenter."
+            )
+        self.augmenter = augmenter or preprocessing.Augmenter(
```

Comment:
I'd rather drop the default augmentor than require include_rescaling.  

It's too confusing when you are also setting it with your encoder.